### ISSUE TITLE
Authentication for portal and routing

### ISSUE THREAD

**Issue by zinfin** [Esri/maps-app-android#149]
Question:  If the user starts the app, signs in to the http:///www.arcgis.com portal with credentials then requests a route they are not prompted for credentials a second time.  But the reverse is not true - if a user starts the app and first requests a route (where they are prompted for credentials) and THEN signs in to the portal to view their content they ARE prompted a second time for credentials.  Why?

**Comment by zinfin** [comment]
@doneill Is this behavior expected to change in upcoming updates or releases?

**Comment by doneill** [comment]
@zinfin Yes, I think this workflow should be updated such that it supports a single sign-on.

**Comment by zinfin** [comment]
The problem:  **A user is forced to authenticate twice if they want to see their portal content and if they want to get a route.**

Solution:  Use a single sign in model by leveraging CredentialCache once the user is logged in (either to Portal or via routing workflow).  This will also enable social login to work correctly.  Currently, once the social login is cached it conflicts with the second attempt to login.
